### PR TITLE
Linters using npm

### PR DESCRIPTION
**Pull request by ElPaisano**
This PR:

1. Adds scripts and logic to run pre-commit spelling, MD formatting, and broken link checks
1. Adds two helper scripts to make the error fix process easier
1. Creates initial README entry https://github.com/filecoin-project/filecoin-docs/blob/fb97b97c0c61e107beaf85a41a049bbb20a86781/README.md#pre-commit-linters
1. Adds initial spelling and MD formatting configs for checks
1. Fixes a bunch of issues identified by the linters

> @johnnymatthews just a thought that we might want to not include step 5 in this, in case our fixes actually break things. Not that they should, but this set up is still untested in the wild, so might be better for our sanity. Could just restore any .md files that aren't the README

TODOS:
- [x] @johnnymatthews pull down and test latest
- [x] @ElPaisano pull down and test latest 
- [ ] Custom markdown-link-check config? -> for the moment, no. Can revisit if need be
- [x] Determine where we want the `.sh` scripts to live. They are currently in root
- [x] Finalize script "experience" and wording, README wording

### CODE REVIEW

File: content/en/basics/the-blockchain/proofs/index.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ In Filecoin, this process is known as _Proof-of-Storage_, and consists of two di
 
 In the Filecoin storage lifecycle process, _Proof-of-Replication (PoRep)_ is used when an SP agrees to store data on behalf of a client and receives a piece of client data. In this process:
 
-1. The data is placed into a _{{< tooltip "sector" >}}_. 
+1. The data is placed into a _{{< tooltip "sector" >}}_.
```
Comment (johnnymatthews) [inline content/en/basics/the-blockchain/proofs/index.md:46]:
```suggestion
1. The data is placed into a {{< tooltip "sector" >}}.
```

**Comment by ElPaisano** [comment]
@johnnymatthews the link check flagged the following issues in README, so I lumped them into this PR to cus why not? 😓 

  ERROR: 7 dead links found in README.md !
  [✖] https://github.com/filecoin-project/filecoin-docs/blob/master/LICENSE.txt → Status: 404
  [✖] #code-tabs → Status: 404
  [✖] http://localhost:1313 → Status: 0
  [✖] ./LICENSE-MIT → Status: 400
  [✖] ./LICENSE-APACHE → Status: 400
  [✖] /.static/images/front-matter-variables-sub-menus.png → Status: 400
  [✖] /.static/images/archived-page.png → Status: 400
  
  LMK if any of those shouldn't have been changed or removed

**Review by johnnymatthews — APPROVED** [review]
This is great! As for where the scripts should go, we can either leave them in the root folder, or throw them into a scripts folder. I don't really mind either way.

File: pre-commit-lint.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        echo " "
+        echo $errorBar
+        echo "${bold}ERRORS FOUND${normal}"
+        echo "There are some problems with your commit:"
+        echo "$errorDescr"
+        echo " "
+        echo "Check $DOCLINK for details on how to fix these errors."
+        echo $errorBar
+        echo " "
+        exit 0
```
[the remark this answers is not in the document]
Comment (ElPaisano) [reply pre-commit-lint.sh]: So I think that, for the moment, this needs to remain set to `1`, as it appears that the  script's `0` exit overrides pre-commits `1`, which causes the pre-commit to allow the commit even when the tests fail. I agree that the extra output is unwanted, but I'm not sure how to fix that ATM. IDK if this is a blocker

**Comment by ElPaisano** [comment]
> This is great! As for where the scripts should go, we can either leave them in the root folder, or throw them into a scripts folder. I don't really mind either way.

Cool, I say lets just keep them in root for now

File: pre-commit-lint.sh [hunk 3]
```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+bold=$(tput bold)
+normal=$(tput sgr0)
+regularBar="----------------------------------------------------"
+boldBar="===================================================="
+errorBar="!**************************************************!"
```
[the remark this answers is not in the document]
Comment (ElPaisano) [reply pre-commit-lint.sh:8]: Very nice,  output looks way better

**Comment by ElPaisano** [comment]
Ok, I've tested and added in the doclink. @johnnymatthews updated script layout looks 💯 . So, remaining items: 

1. Exit code stuff https://github.com/filecoin-project/filecoin-docs/pull/1972#pullrequestreview-1410909196
1. Made trivial teaks to the script and README update

Other than that, I think this is good to go

**Comment by johnnymatthews** [comment]
There. I've changed the branch that this PR is going into. Once the pre-commit stuff is ready, let's merge it in. Then we can run the pre-commit checks against everything in the repo.

This way we can build upon the spelling-list/dictionary, and fix any funking formatting before external contributors have to.

**Pull request merged**